src/NzbDrone.Core/Download/Clients/qBittorrent/QBittorrentProxy.cs

```diff
@@ -0,0 +1,187 @@
+            return response.Read<Dictionary<String, Object>>(client);
+        }
+
+        public List<QBittorrentTorrent> GetTorrents(QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/query/torrents", Method.GET);
+            request.RequestFormat = DataFormat.Json;
+            request.AddParameter("label", settings.TvCategory);
+
+            var client = BuildClient(settings);
+            var response = ProcessRequest(client, request, settings);
+            return response.Read<List<QBittorrentTorrent>>(client);
+        }
+
+        public void AddTorrentFromUrl(String torrentUrl, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/command/download", Method.POST);
+            request.AddParameter("urls", torrentUrl);
+
+            var client = BuildClient(settings);
+            ProcessRequest(client, request, settings);
+        }
+
+        public void AddTorrentFromFile(String fileName, Byte[] fileContent, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/command/upload", Method.POST);
+            request.AddFile("torrents", fileContent, fileName);
+
+            var client = BuildClient(settings);
+            ProcessRequest(client, request, settings);
+        }
+
+        public void RemoveTorrent(String hash, Boolean removeData, QBittorrentSettings settings)
+        {
+            String cmd = removeData ? "/command/deletePerm" : "/command/delete";
+            var request = new RestRequest(cmd, Method.POST);
+            request.AddParameter("hashes", hash);
+
+            var client = BuildClient(settings);
+            ProcessRequest(client, request, settings);
+        }
+        public void SetTorrentLabel(String hash, String label, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/command/setLabel", Method.POST);
+            request.AddParameter("hashes", hash);
+            request.AddParameter("label", label);
+
+            var client = BuildClient(settings);
+            ProcessRequest(client, request, settings);
+        }
+
+        public void MoveTorrentToTopInQueue(String hash, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/command/topPrio", Method.POST);
+            request.AddParameter("hashes", hash);
+
+            var client = BuildClient(settings);
+            ProcessRequest(client, request, settings);
+        }
+
+        private IRestResponse ProcessRequest(IRestClient client, IRestRequest request, QBittorrentSettings settings)
+        {
+            _logger.Debug("Url: {0}", client.BuildUri(request));
+            var clientResponse = client.Execute(request);
+
+            if (clientResponse.StatusCode == HttpStatusCode.Forbidden)
+            {
+                _logger.Debug("QBittorrent authentication error.");
+
+                Login(client, settings);
```

Comment:
This will trigger a login every request, instead we can cache that we authenticated correctly (same as we do for Plex Server).